line_obfuscator.go

```diff
@@ -68,21 +110,42 @@ func clearNodeComments(node ast.Node) {
 	}
 }
 
-func findBuildTags(commentGroups []*ast.CommentGroup) (buildTags []string) {
-	for _, group := range commentGroups {
-		for _, comment := range group.List {
-			if !strings.Contains(comment.Text, "+build") {
+func processSpecialComments(commentGroups []*ast.CommentGroup) (extraComments, localnameBlacklist []string) {
+	var buildTags []string
+	var specialComments []string
+	for _, commentGroup := range commentGroups {
+		for _, comment := range commentGroup.List {
+			if strings.HasPrefix(comment.Text, buildTagPrefix) {
+				buildTags = append(buildTags, comment.Text)
+				continue
+			}
+
+			if !isDirective(comment.Text, specialDirectives) {
 				continue
 			}
-			buildTags = append(buildTags, comment.Text)
+
+			specialComments = append(specialComments, comment.Text)
+			if localName, ok := getLocalName(comment.Text); ok {
+				localnameBlacklist = append(localnameBlacklist, localName)
+			}
 		}
 	}
-	return buildTags
+
+	extraComments = append(extraComments, buildTags...)
+	extraComments = append(extraComments, specialComments...)
+	extraComments = append(extraComments, "")
+	return extraComments, localnameBlacklist
 }
 
-func transformLineInfo(file *ast.File) ([]string, *ast.File) {
+func transformLineInfo(file *ast.File, cgoFile bool) ([]string, []string, *ast.File) {
```

Comment:
name the return parameters; `[]string, []string` tells me nothing.

also, this func is getting more complex, so it needs a godoc now.